actions/workflow-lint/action.yml

```diff
@@ -0,0 +1,5 @@
+name: D2L GitHub Action Workflow Linter
```

Comment:
*Why a separate action?* we could probably do it just with a `run:` step, but then because this template will get copy-pasted into repos we are asking for drift.

*Why define the action in this repo?* I don't feel strongly about this. This action is meant to be used with this template file, so I felt it maybe didn't make sense to put in it's own repo. It's the kind of action that I don't think is going to be (directly) useful outside of our org (the way I intend to write it).